src/gui/lineedit/src/lineedit.cpp

```diff
@@ -8,48 +8,73 @@
-  clearButton = new QToolButton(this);
-  QPixmap pixmap2(":/lineeditimages/clear_left.png");
-  clearButton->setIcon(QIcon(pixmap2));
-  clearButton->setIconSize(pixmap2.size());
-  clearButton->setCursor(Qt::ArrowCursor);
-  clearButton->setStyleSheet("QToolButton { border: none; padding: 2px; }");
-  clearButton->setToolTip(tr("Clear the text"));
-  clearButton->hide();
-  connect(clearButton, SIGNAL(clicked()), this, SLOT(clear()));
-  connect(this, SIGNAL(textChanged(const QString&)), this, SLOT(updateCloseButton(const QString&)));
-  int frameWidth = style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
-  setStyleSheet(QString("QLineEdit { padding-right: %1px; padding-left: %2px; }").arg(clearButton->sizeHint().width() + frameWidth + 1).arg(clearButton->sizeHint().width() + frameWidth + 1));
-  QSize msz = minimumSizeHint();
-  setMinimumSize(qMax(msz.width(), clearButton->sizeHint().width() + searchButton->sizeHint().width() + frameWidth * 2 + 2),
-                 qMax(msz.height(), clearButton->sizeHint().height() + frameWidth * 2 + 2));
+    int frameWidth = style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
+
+    QPixmap pixmap1(":/lineeditimages/search.png");
+    searchButton = new QToolButton(this);
+    searchButton->setIcon(QIcon(pixmap1));
+    searchButton->setIconSize(pixmap1.size());
+    searchButton->setCursor(Qt::ArrowCursor);
+    searchButton->setStyleSheet("QToolButton { border: none; padding: 2px; }");
+
+    int clearButtonSizeHintWidth = 0;
+    int clearButtonSizeHintHeight = 0;
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
+    QPixmap pixmap2(":/lineeditimages/clear_left.png");
+    clearButton = new QToolButton(this);
+    clearButton->setIcon(QIcon(pixmap2));
+    clearButton->setIconSize(pixmap2.size());
+    clearButton->setCursor(Qt::ArrowCursor);
+    clearButton->setStyleSheet("QToolButton { border: none; padding: 2px; }");
+    clearButton->setToolTip(tr("Clear the text"));
+    clearButton->hide();
+    connect(clearButton, SIGNAL(clicked()), this, SLOT(clear()));
+    connect(this, SIGNAL(textChanged(const QString &)), this, SLOT(updateCloseButton(const QString &)));
+
+    clearButtonSizeHintWidth = clearButton->sizeHint().width();
+    clearButtonSizeHintHeight = clearButton->sizeHint().height();
+    setStyleSheet(QString("QLineEdit { padding-left: %1px; padding-right: %2px; }").arg(searchButton->sizeHint().width()).arg(clearButtonSizeHintWidth));
+#else
+    setClearButtonEnabled(true);
+    setStyleSheet(QString("QLineEdit { padding-left: %1px; }").arg(searchButton->sizeHint().width()));  // padding between text and widget borders
+#endif
+
+    QSize msz = sizeHint();
+    setMinimumSize(qMax(msz.width(), searchButton->sizeHint().width() + clearButtonSizeHintWidth),
+                   std::max({msz.height(), searchButton->sizeHint().height(), clearButtonSizeHintHeight}) + frameWidth * 2);
 }
 
-void LineEdit::resizeEvent(QResizeEvent *)
+LineEdit::~LineEdit()
```

Comment:
Once we create the button with **this** as parent, we don't need this destructor (QObject destructor removes them).